### PR TITLE
Add Thai word list from Thai Wikipedia titles

### LINKED ISSUE #858: [Suggestion] Add a large dictionary data
## Detailed description
I would suggest adding a large (300K+ sized) dictionary for better tokenization performance.

## Context
I am currently doing text mining of Pantip. Pantip has a lot of new(?) words and proper-nouns that `pythainlp.corpus.common.thai_words()` couldn't catch.
But when I added new words from
* [Volubilis Dictionary](https://belisan-volubilis.blogspot.com/) for new words
* [Titles of wikipedia-Th articles](https://dumps.wikimedia.org/thwiki/latest/thwiki-latest-all-titles-in-ns0.gz) for proper-nouns

and the performance improved by around 10%. The dictionary became 300K words in total.

I guess that it could be useful if this large dictionary data is easily (just import from pythainlp modules) available for other users too.

## Possible implementation
Simply make a dictionary data from the sources above and serve it as `pythainlp.corpus.common.thai_words_large()` or something (Since dynamically downloading from the sources above could be a burden for the providers)

### PR DESCRIPTION
**Pull request by konbraphat51**
### What does this changes

Add an optional corpus of Wikipedia titles.
* thai_wikipedia_titles() function
* license description

Fixes #858 

### Your checklist for this pull request
🚨Please review the [guidelines for contributing](https://github.com/PyThaiNLP/pythainlp/blob/dev/CONTRIBUTING.md) to this repository.

- [x] Passed code styles and structures
- [x] Passed code linting checks and unit test

### CODE REVIEW

**Comment by konbraphat51** [comment]
Review please

**Comment by bact** [comment]
Looks good to me.

File: tests/test_corpus.py [hunk 1]
```diff
@@ -25,6 +25,7 @@
     thai_syllables,
     thai_synonym,
     thai_words,
+    wikipedia_titles,
```
Comment (bact) [inline tests/test_corpus.py]: Do you mind to put it in alphabetical order please?

**Review by bact — CHANGES_REQUESTED** [review]
Code, docs, license, test. All looks good.

Once sorted out the order of module imports, I can approve this.

File: pythainlp/corpus/corpus_license.md [hunk 2]
```diff
@@ -101,3 +101,34 @@ S. Thoongsup et al., ‘Thai WordNet construction’,
 in Proceedings of the 7th Workshop on Asian Language Resources,
 Suntec, Singapore, Aug. 2009, pp. 139–144.
 https://www.aclweb.org/anthology/W09-3420.pdf 
+
+## Wikipedia Titles
+Corpus of Wikipedia titles (wikipedia_titles.txt) was processed by konbraphat51 (https://github.com/konbraphat51/Thai_Dictionary_Cleaner/tree/main)
+
+The original data is thwiki-latest-all-titles.gz of https://dumps.wikimedia.org/thwiki/latest/ which Wikipedia.org has created.
```
Comment (bact) [inline pythainlp/corpus/corpus_license.md]: Do you mind to add date of the Wikipedia data here please?
The date that you have downloaded the data for your preparation.

File: tests/test_corpus.py (hunk 1 above)
Comment (konbraphat51) [reply tests/test_corpus.py]: Oh sorry, fixed!

File: pythainlp/corpus/corpus_license.md (hunk 2 above)
Comment (konbraphat51) [reply pythainlp/corpus/corpus_license.md]: Yeah that's definitely required. Added!

**Comment by konbraphat51** [comment]
Okey, review again please

**Review by wannaphong — APPROVED** [review]
It look great!

**Review by bact — APPROVED** [review]
Thank you.

**Comment by konbraphat51** [comment]
Ok, handled with the conflict!

**Review by bact — APPROVED** [review]

**Comment by bact** [comment]
Merged. Thank you.

**Pull request merged**